### DISCUSSION TITLE
the category lazy example cause errors

### DISCUSSION THREAD

**Discussion by naivefun** [colinhacks/zod#612] [Q&A]
```
interface Category {
  name: string;
  subcategories: Category[];
}

// cast to z.ZodSchema<Category>
const Category: z.ZodSchema<Category> = z.lazy(() =>
  z.object({
    name: z.string(),
    subcategories: z.array(Category),
  })
);
```

![image](https://user-images.githubusercontent.com/10683378/130912889-4b26d9fb-a3d0-4c75-b20c-c8233cb5711e.png)

Seems lazy returns all optional interface but it is not optional in self-defined Category. How to fix this?

**Comment by naivefun** [comment]
![image](https://user-images.githubusercontent.com/10683378/130917198-d19b432b-c444-4adf-a3de-918813a61b7a.png)
Seems the basic infer example also lost required information, is that normal?

**Comment by scotttrinh** [comment]
@naivefun Is it possible that there is something going on with your editor setup, or maybe your `tsconfig`? Are you running `"strict": true` in your `tsconfig.json`? If not, that might be your issue.

Works on [TypeScript playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgLzgXzgMyhEcBEyEAJvgNwBQFwAdjAKZSYCGAxvXAMLMMDm0AT0QU4cGsxD0AXHADOMKLV6VRsgK4AjVj3r9F9WTO59BAbQC6lNFQD0NuNvlwYEFADoAWiQDKrABb0IMwAPMa6ggB8FKwQNE5hegIyyJ4+-oEhCZFwALzuADbMyAIAFCUAlLlRoikQGgBW9KwwJQgiomIS0u7yijS8FQA07aqa2ib6hu7MUFDMpVlQAuXDomjlFOWUQA):

<img width="655" alt="image" src="https://user-images.githubusercontent.com/1682194/131056437-442e5d1a-9714-4100-bc28-2b83b9cacd01.png">
